ext/oci8/oci8_interface.c

```diff
@@ -44,6 +44,29 @@
 #define OCI_STMT_CALL 10
 #endif
 
+/* {{{ proto bool oci_register_taf_callback( resource connection [, string callback] )
+   Register a callback function for Oracle Transparent Application Failover (TAF) */
+PHP_FUNCTION(oci_register_taf_callback)
+{
+	zval *z_connection;
+	php_oci_connection *connection;
+	char *callback = NULL;
+	int callback_len = 0;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "r|s!", &z_connection, &callback, &callback_len) == FAILURE) {
```

Comment:
I don't have a great answer here -- fcall_info is a bit of a mess. The simplest way to handle this is probably to not bother: Instead accept a zval, use zend_is_callable to make sure it's callable and store that zval (with ZVAL_COPY). Then pass it to call_user_function the way you do now. As you say, this callback will not be commonly invoked, so there is no performance advantage to properly managing fci/fcc here.